File: programmers/js/12. 제일 작은 수 제거하기.js
function solution(arr) {
    const answer = [];

    if(arr.length == 1) {
        answer.push(-1);
        return answer;
        
    } else {
        const min = Math.min(...arr); // 배열에서 최솟값
        // console.log('최솟값 : ', min);
        // console.log('최솟값 인덱스 : ', arr.indexOf(min))
        
        arr.splice(arr.indexOf(min),1);

        return arr;
    }   
}

console.log(solution([-10]));
console.log(solution([6,5,1,2]));
console.log(solution([9,3,6,5,10,22]));


// var a = [0, 1, 2, 3, 4, 5];
// 2번 배열부터 3개 제거.
// var b = a.splice(2, 3);

//console.log(a); //  [0, 1, 5]
//console.log(b); //  [2, 3, 4]

File: programmers/java/12. 제일 작은수 제거하기.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr) {
        int[] answer = {};
        ArrayList<Integer> arrList = new ArrayList<>();
        
        if(arr.length == 1) { // 1이면 -1
            return new int[]{-1};
        }
        
        for(int a : arr) {
            arrList.add(a);
        }
        
        int min = arr[0];
        for(int a : arrList) { // 최솟값
            if(a < min) {
                min = a;
            }
        }
        
        arrList.remove(arrList.indexOf(min)); // 최솟값 제거
        
        if(arrList.size() > 1) {
            answer = new int[arrList.size()];
            for( int i=0; i < arrList.size(); i++) {
                answer[i] = arrList.get(i);
            }
        }
        
        return answer;
    }
}

//  다른사람
//  if (arr.length <= 1) return new int[]{ -1 };
//       int min = Arrays.stream(arr).min().getAsInt();
//       return Arrays.stream(arr).filter(i -> i != min).toArray();